app/controllers/application_controller.rb

```diff
@@ -24,6 +24,10 @@ class ApplicationController < ActionController::Base
   # See the `pretender` gem.
   impersonates :user
 
+  def signed_in_and_confirmed?
+    signed_in? && current_user.try(:confirmed?)
```

Comment:
Does `current_user` not being `nil` not already tell the same as `signed_in?`

So I'd say
```ruby
def signed_in_and_confirmed?
  current_user&.confirmed?
end
```
Does the same thing but shorter.

Reply:
Not sure if that's pedantic (feel free to ignore me), but relying on `signed_in?` == `current_user != nil` is digging into implementation details and lays a trap if we one day decide to e.g. have current_user always return a user-like object (like a `GuestUser` decorator).